web/components/tipper.tsx

```diff
@@ -67,30 +59,75 @@ export function Tipper(prop: {
         fromId: user.id,
         toId: comment.userId,
       })
-    }, 1500)
+    }
   )
-  // instant save on unrender
-  useEffect(() => () => void saveTip.flush(), [saveTip])
 
   const addTip = (delta: number) => {
-    setLocalTip(localTip + delta)
-    me && saveTip(me, comment, localTip - myTip + delta)
-    toast(`You tipped ${comment.userName} ${formatMoney(LIKE_TIP_AMOUNT)}!`)
+    setUserTipped(true)
+    const timeoutId = setTimeout(() => {
+      me && saveTip(me, comment, delta)
+    }, 3000)
+    toast.custom(
+      () => (
+        <TipToast
+          userName={comment.userName}
+          onUndoClick={() => {
+            clearTimeout(timeoutId)
+          }}
+        />
+      ),
+      { duration: TIP_UNDO_DURATION }
+    )
   }
 
   const canUp =
-    me && comment.userId !== me.id && me.balance >= localTip + LIKE_TIP_AMOUNT
+    me && comment.userId !== me.id && me.balance >= myTip + LIKE_TIP_AMOUNT
```

Comment:
I think you don't want to add `myTip` here. That's all the previous tips that already went through